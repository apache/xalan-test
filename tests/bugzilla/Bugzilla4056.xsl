<?xml version="1.0" encoding="UTF-8"?> 
<!-- Bugzilla#4056 See also whitespace20.xsl for similar test of xml:space 

jkesselm May2023 note: Appears to be functioning correctly at this time.
jkesselm May2023 note: Appears to be functioning correctly at this time.
We could make it one of the whitespace regression tests if desired, or
a contrib/ test, but I'm not sure I see a need to do so.
Close as resolved?
-->
<!-- $RCSfile: Bugzilla4056.xsl,v $ $Revision$ $Date$ -->
<xsl:stylesheet 
 xmlns:xsl="http://www.w3.org/1999/XSL/Transform"  
 version="1.0"
  xmlns:java="http://xml.apache.org/xslt/java"
  exclude-result-prefixes="java">
  <xsl:strip-space elements="*"/>

  <xsl:template match="NewsML">
	<xsl:apply-templates select="NewsItem"/>
  </xsl:template>

  <xsl:template match="NewsItem">
    <EVENT xml:space="preserve">
	  <xsl:attribute name="event-id"><xsl:value-of select="Identification/NewsIdentifier/PublicIdentifier"/>:<xsl:value-of select="Identification/NewsIdentifier/RevisionId"/></xsl:attribute>
	  <DIR name="content">
		<STRING name="headline">
		  <xsl:choose>
			<xsl:when test="string-length( NewsComponent/NewsLines/HeadLine )>0">
			  <xsl:apply-templates select="NewsComponent/NewsLines/HeadLine/descendant-or-self::text()"/>
		  </xsl:when>
		  <xsl:otherwise>
			<xsl:apply-templates select="NewsComponent/NewsLines/NewsLine[NewsLineType/@FormalName='alert'][1]/NewsLineText/descendant-or-self::text()"/>
		  </xsl:otherwise>
		</xsl:choose>
		</STRING>
		<STRING name="story">
		  <xsl:apply-templates select="NewsComponent/ContentItem/DataContent/descendant-or-self::text()"/>
		</STRING>
	  </DIR>
	  <DIR name="meta">
		<xsl:apply-templates select="NewsComponent/TopicSet[@FormalName='ReutersMetaData']/Topic/Property[@FormalName='Code']" mode="meta"/>
		
		<STRING name="provider"><xsl:value-of select="Identification/NewsIdentifier/ProviderId"/></STRING>
		<STRING name="any"><xsl:value-of select="Identification/NewsIdentifier/ProviderId"/></STRING>
		
		<xsl:variable name="attribution" select="NewsComponent/AdministrativeMetadata/Source/Party/@FormalName"/>
		<xsl:if test="$attribution">
		  <STRING name="attribution"><xsl:value-of select="$attribution"/></STRING>
		  <STRING name="any"><xsl:value-of select="$attribution"/></STRING>
		</xsl:if>

		<xsl:variable name="language" select="NewsComponent/DescriptiveMetadata/Language[1]/@FormalName"/>
		<xsl:if test="$language">
		  <STRING name="language"><xsl:value-of select="$language"/></STRING>
		  <STRING name="any"><xsl:value-of select="$language"/></STRING>
		</xsl:if>
	  </DIR>
	  <DIR name="teasers">
		<STRING name="teaser0"/>
		<STRING name="teaser1"/>
		<STRING name="teaser2"/>
		<STRING name="teaser3"/>
		<STRING name="teaser4"/>
	  </DIR>
	</EVENT>
  </xsl:template>

  <xsl:template match="Property" mode="meta">
	<STRING name="{../TopicType/@FormalName}"><xsl:value-of select="@Value"/></STRING>
	<STRING name="any"><xsl:value-of select="@Value"/></STRING>
  </xsl:template>

  <!--
   * Licensed to the Apache Software Foundation (ASF) under one
   * or more contributor license agreements. See the NOTICE file
   * distributed with this work for additional information
   * regarding copyright ownership. The ASF licenses this file
   * to you under the Apache License, Version 2.0 (the  "License");
   * you may not use this file except in compliance with the License.
   * You may obtain a copy of the License at
   *
   *     http://www.apache.org/licenses/LICENSE-2.0
   *
   * Unless required by applicable law or agreed to in writing, software
   * distributed under the License is distributed on an "AS IS" BASIS,
   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   * See the License for the specific language governing permissions and
   * limitations under the License.
  -->

</xsl:stylesheet>
