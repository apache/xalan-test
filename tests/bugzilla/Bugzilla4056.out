<?xml version="1.0" encoding="UTF-8"?><EVENT xml:space="preserve">
	  
	  <DIR name="content">
		<STRING name="headline">
		  
			  NZx Morning Call-NZ ADRs flat, Telecom  TEL.NZ  slips
		  
		</STRING>
		<STRING name="story">
		      WELLINGTON, Oct 10 (Reuters) - Prices of 
selected New Zealand 
stocks listed in the United States, and the change from their 
close on the previous New Zealand and U.S. trading days. 
    All columns are shown converted to New Zealand shares and 
currency except the first column which shows the ADR priced in 
U.S. dollars: 
                         ADR      -----------Converted-------- 
                        Price     Price    Change vs    Volume 
 Company:                USD       NZD     NZ close      (000) 
 FCL Building FLB.N     11.16      2.71      
nil           30  
 FCL Forests  FFS.N      1.00      0.24      
nil          115  
 Telecom      NZT.N     14.28      4.33       -
4          125  
 Tranz Rail  TNZR.O      untraded 
    NOTE- The above rates were at 8:00 a.m. (NZT) when the Dow 
Jones Industrial Average  .DJI  was down 23 
points at 9,045. They 
are based on a conversion rate of NZ$=$0.4124. 
    On the New York Stock Exchange, Telecom American Depositary 
Receipts are traded in bundles of eight ordinary shares, while 
Fletcher Building and Forests shares are traded in bundles of 10. 
    On the Nasdaq exchange, Tranz Rail ADSs are traded in bundles 
of three ordinary shares. 
    ((Wellington newsroom, tel +64 +4 471-4277, fax +64 +4 473 
6212, wellington.newsroom@reuters.com)) 

		</STRING>
	  </DIR>
	  <DIR name="meta">
		<STRING name="providedby">NDS</STRING><STRING name="any">NDS</STRING><STRING name="companies">.NZ40</STRING><STRING name="any">.NZ40</STRING><STRING name="companies">FBU.NZ</STRING><STRING name="any">FBU.NZ</STRING><STRING name="companies">FFS.NZ</STRING><STRING name="any">FFS.NZ</STRING><STRING name="companies">TEL.NZ</STRING><STRING name="any">TEL.NZ</STRING><STRING name="companies">TRH.NZ</STRING><STRING name="any">TRH.NZ</STRING><STRING name="nameditem">NZ/ADR</STRING><STRING name="any">NZ/ADR</STRING><STRING name="topics">NZ</STRING><STRING name="any">NZ</STRING><STRING name="topics">ASIA</STRING><STRING name="any">ASIA</STRING><STRING name="topics">STX</STRING><STRING name="any">STX</STRING><STRING name="topics">US</STRING><STRING name="any">US</STRING><STRING name="topics">LEN</STRING><STRING name="any">LEN</STRING><STRING name="topics">RTRS</STRING><STRING name="any">RTRS</STRING><STRING name="products">ND</STRING><STRING name="any">ND</STRING><STRING name="products">E</STRING><STRING name="any">E</STRING><STRING name="products">RNP</STRING><STRING name="any">RNP</STRING><STRING name="products">DNP</STRING><STRING name="any">DNP</STRING><STRING name="products">PCO</STRING><STRING name="any">PCO</STRING><STRING name="products">PCU</STRING><STRING name="any">PCU</STRING><STRING name="headline_pes">511 376 408 416 433 443 458</STRING><STRING name="any">511 376 408 416 433 443 458</STRING><STRING name="story_pes">511 376 408 416 433 443 458</STRING><STRING name="any">511 376 408 416 433 443 458</STRING>
		
		<STRING name="provider">reuters.com</STRING>
		<STRING name="any">reuters.com</STRING>
		
		
		
		  <STRING name="attribution">RTRS</STRING>
		  <STRING name="any">RTRS</STRING>
		

		
		
		  <STRING name="language">EN</STRING>
		  <STRING name="any">EN</STRING>
		
	  </DIR>
	  <DIR name="teasers">
		<STRING name="teaser0"/>
		<STRING name="teaser1"/>
		<STRING name="teaser2"/>
		<STRING name="teaser3"/>
		<STRING name="teaser4"/>
	  </DIR>
	</EVENT>